File: java/sort/h_index.java


import java.util.*;

/*
[3, 0, 6, 1, 5]	3

[0, 1, 3, 5, 6]

*/

class Solution {
    public int solution(int[] citations) {       
        int h_index = 0;
        Arrays.sort(citations);

        ArrayList<Integer> arr = new ArrayList<Integer>();

        for (int i=0; i<citations.length; i++) {
            arr.add(citations[i]);
        }

        for (int i=0; i<arr.size(); i++) {
            int c = arr.get(arr.size() - 1 - i);
            if (c > h_index) {
                h_index ++;
            }
        }

        return h_index;
    }
}

File: cpp/h-index.cpp
// H-Index
// https://school.programmers.co.kr/learn/courses/30/lessons/42747?language=cpp

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int hindex = 0;
    sort(citations.rbegin(), citations.rend());

    for (int c : citations) {
        if (c > hindex) {
            hindex ++;
        }
    }
    return hindex;
}

int main(int argc, char** argv)
{
    vector<int> citations = {3, 0, 6, 1, 5};
    cout << solution(citations) << endl;

    return 0;
}